### PR TITLE
feat(code-editor): Dynamic code editor in the demo

### PR DESCRIPTION
**Pull request by GermainBergeron**
### Proposed Changes

Required code to have a Sandbox component that accept any code and render the results

https://user-images.githubusercontent.com/260007/146600870-0fdc64e3-65c5-4889-8011-e1a73b68f2c8.mov

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by GermainBergeron** [comment]
I had to put a few `as any` in the code since we're on an older version of typescript and the dependencies are resolved in a weird way. `@typescript/vfs` declare its typescript dependency as `*` which resolves to the one installed by the design token package

**Comment by aiheon** [comment]
> I had to put a few `as any` in the code since we're on an older version of typescript and the dependencies are resolved in a weird way. `@typescript/vfs` declare its typescript dependency as `*` which resolves to the one installed by the design token package

I suppose we will be able to remove them soon enough with the bump of typescript for the search bar :smile:

**Review by aiheon — APPROVED** [review]

**Comment by gdostie** [comment]
> which resolves to the one installed by the design token package

I got a few errors fixed by changing the typescript version to 3.8.3 in the design-tokens package

File: packages/demo/src/demo-building-blocs/useTypescriptServer.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            createDefaultMapFromCDN(compilerOptions as any, ts.version!, true, ts as any, lzstring).then((map) => {
+                mappedTypes.forEach(({path, content}) => {
+                    monaco.languages.typescript.typescriptDefaults.addExtraLib(content, `file://${path}`);
+                    map.set(path, content);
+                });
+                setFsMap(map);
+            });
+        });
+    }, []);
+    return {fsMap, setFsMap, compilerOptions};
```
Comment (gdostie) [inline packages/demo/src/demo-building-blocs/useTypescriptServer.ts:60]: When implementing the props table, I realized that the setFsMap doesn't need to be exposed. To add stuff to the file system we can just do `fsMap.set`

```suggestion
    return {fsMap, compilerOptions};
```

**Review by gdostie — APPROVED** [review]

**Pull request merged**